reppy/cache/__init__.py

```diff
@@ -65,7 +66,13 @@ def factory(self, url):
         try:
             return self.fetch(url)
         except BaseException as exc:
-            logger.exception('Reppy cache fetch error on %s' % url)
+            # Log the whole stack trace only if the exception is anything other than a ReadTimeout
+            # or if we do not have a default factory
+            if isinstance(exc, ReadTimeout) and isinstance(self.cache_policy, DefaultObjectPolicy):
```

Comment:
Probably not a good way to test it, but to at least exercise this code path, in `test_cache.py:TestBaseCache`, you can add something similar to some of the other tests to be along the lines of:

```python
cache = cache.BaseCache(10)
with mock.patch.object(cache, 'fetch', side_effect=ReadTimeout()):
    with self.assertRaises(ReadTimeout):
        cache.factory('http://some-url')
```

The only reason that Travis appears to be failing is to do with line and branch coverage. Oddly enough, that seems to only be the case in certain Python versions

with self.assertRaises(NotImplementedError):
            cache.BaseCache(10).fetch('http://example.com/robots.txt')

Reply:
I just updated my commit with the proper tests, thanks for your guidance.